pages/partials/countdown.ejs

```diff
@@ -49,23 +52,43 @@ process) in milliseconds as the variable "serverRemainingMS".
 
 <script>
   window.allowQuestionUnload = false;
-  var clientStart;
-  if (window.performance && window.performance.timing && window.performance.timing.responseStart) {
-      clientStart = window.performance.timing.responseStart;
-  } else {
-      clientStart = Date.now();
-  }
+  var serverRemainingMS = <%= serverRemainingMS %>;
+  var serverTimeLimitMS = <%= serverTimeLimitMS %>;
+  var clientStart = Date.now();
+  
   var countdownDisplay = $('<%= displaySelector %>');
   var countdownProgress = $('<%= progressSelector %>');
   countdownProgress.addClass('progress');
   countdownProgress.html('<div class="progress-bar progress-bar-primary"></div>');
   countdownProgressBar = countdownProgress.children('div');
 
+  if (serverRemainingMS <= 0) {
+      window.allowQuestionUnload = true;
+      document.forms['time-limit-finish-form'].submit();
+  }
+
+  var handleServerResponseRemainingMS = function(data, textStatus, jqxhr) {
+      serverRemainingMS = data.serverRemainingMS;
+      serverTimeLimitMS = data.serverTimeLimitMS;
+      console.log('Time remaining: ' + serverRemainingMS + ' ms');
+      clientStart = Date.now();
+      if (serverRemainingMS <= 500) {
```

Comment:
Should we change this to give them slightly more time? We'd also need to change `remainingMS > 500` below to `remainingMS > 0`.
```suggestion
      if (serverRemainingMS <= 0) {
```